OpenRA.Mods.Common/ActorExts.cs

```diff
@@ -18,6 +18,22 @@ namespace OpenRA.Mods.Common
 {
 	public static class ActorExts
 	{
+		public static bool IsAtGroundLevel(this Actor self)
```

Comment:
Should this check `self.IsInWorld`?

Upon thinking over this I've come to the conclusion that just testing this won't be enough.
Logic sets will actually want to know the difference between the terrain height and the location of the actor in question.
This goes for ground level, under(ground|water), airborne.

We could add an enum for these cases:

```csharp
enum TerrainRelativePosition
{
    BelowTerrain,
    AtTerrain,
    AboveTerrain
}
```

And build helper methods off of that (which would have to be decided using `Map::AltitudeAboveTerrain` and `Map::TerrainHeightAt`.